oauth_dropins/flickr_auth.py

```diff
@@ -72,13 +86,76 @@ def call_api_method(method, params, token_key, token_secret):
 
   # Flickr returns HTTP success even for errors, so we have to fake it
   if body.get('stat') == 'fail':
-    error_code = body.get('code')
-    # https://www.flickr.com/services/api/flickr.auth.checkToken.html#Error%20Codes
-    if error_code == 98 or error_code == 100:
-      code = 401  # invalid auth token or API key -> unauthorized
-    else:
-      code = 400
-    raise urllib2.HTTPError(url, code, 'message=%s, flickr code=%d' % (
-      body.get('message'), error_code), {}, None)
+    raise_for_failure(url, body.get('code'), body.get('message'))
 
   return body
+
+
+def upload(params, photo_file, token_key, token_secret, timeout=None):
```

Comment:
re timeout, yeah, maybe just remove it here and hard-code `appengine_config.HTTP_TIMEOUT`? that's what we do elsewhere.